ipc/common/messenger.go

```diff
@@ -0,0 +1,183 @@
+package common
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/gob"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/ElrondNetwork/arwen-wasm-vm/ipc/logger"
+)
+
+// Messenger intermediates communication (message exchange) via pipes
+type Messenger struct {
+	Name   string
+	Nonce  uint32
+	Logger logger.Logger
+	reader *os.File
+	writer *os.File
+}
+
+// NewMessenger creates a new messenger
+func NewMessenger(name string, logger logger.Logger, reader *os.File, writer *os.File) *Messenger {
```

Comment:
messenger should get marshalizer as well.

Reply:
Since `marshalizer` type agreement between Node and Arwen would increase a bit the complexity, and since Arwen Driver (referenced in the Node) stays in Arwen's packages anyway (at this moment at least), I chose to extract the serialization logic to a separate file, not as a separate component to be injected in constructor (for the moment being). Is this reasonable?

We can also implement the better design when we tune the serialization anyway.

Reply:
the messenger is created by the Arwen driver, so the driver gives the common marshalizer between parent and child.

Reply:
Fixed.